### ISSUE TITLE
Time series panel with bars graph style shows "Data outside time range"

### ISSUE THREAD

**Issue by ivanahuckova** [grafana/grafana#72507]
How to reproduce:
1. Choose test data source as it does not support full range logs volume
2. Add second query with different number of lines (e.g. `10` first and `100` second)
3. It produces incorrect logs volume range that results in "Data outside time range"

<img width="1504" alt="image" src="https://github.com/grafana/grafana/assets/30407135/fb5e5806-db34-412b-ab80-e0e7685442fe">

**Comment by gtk-grafana** [comment]
Taking a look as an onboarding activity.

**Comment by gtk-grafana** [comment]
So this appears to be a consequence of how we're generating the data in the backend of the testdatasource in [scenarios.go](https://github.com/grafana/grafana/blob/main/pkg/tsdb/testdatasource/scenarios.go#L675).

I noticed that the dataframes returning from the backend are in reverse order, with the latest dataframes with the lowest index. 

First I tried to reverse the time values on the backend, but realized my go skills are terrible, wound up [changing the
whole loop](https://github.com/grafana/grafana/blob/gtk-grafana/logs/issues/72507__start-at-from-and-goto-to/pkg/tsdb/testdatasource/scenarios.go#L676).

This gets rid of the data outside range component in the viz, but we're not doing as good of a job simulating a logs query: now all of the results stack up at the beginning instead of being nicely distributed throughout the query range.

<img width="1726" alt="image" src="https://github.com/grafana/grafana/assets/109082771/3ff36324-498b-4ced-82ed-7d2611a15ccd">

And then the next smoking gun: If you take a look at a request with multiple queries when exhibiting this bug, you'll see that the response is much shorter then what you requested, but it still has the right number of logs! Here I'm requesting the last hour, but only 15 minutes are returned!
<img width="1724" alt="image" src="https://github.com/grafana/grafana/assets/109082771/c985036b-46f0-4cf7-acc0-16444a5ce1b3">

It looks like the backend is calculating the expected log lines based on some values from the first query. I wonder if we change the order of our queries, if the bug will go away?

<img width="1709" alt="image" src="https://github.com/grafana/grafana/assets/109082771/b483f780-0ba9-46d8-b196-5aac4598f387">

Neat! So as long as our longest (most lines) query is first, then the range will calculate as we'd expect. 
My next step was to make this change on the backend, I attempted to sort the queries by Lines in the `handleLogsScenario`, but that didn't work.
Setting aside for now, hoping someone more familiar with go could clean this up quickly.

**Comment by ivanahuckova** [comment]
@gtk-grafana we do a lot of logs ordering & calculating visible range within logsModel, so I would hope it should be robust enough that it could handle any results, even if they are incorrectly ordered, or even if time range is outsoide of selected time range. So I would check if there is something we can do better within [logsModel](https://github.com/grafana/grafana/blob/main/public/app/features/logs/logsModel.ts#L266-L286)  or [LogsVolumePanelList](https://github.com/grafana/grafana/blob/main/public/app/features/explore/Logs/LogsVolumePanelList.tsx#L80-L82).

**Comment by gtk-grafana** [comment]
So right now the logs volume panel has no knowledge of what the requested range is, it only calculates internal ranges based on the response data. Which is why currently the volume panel is showing this bug, because the backend is only returning data for the most recent 5% of the graph. 
But the viz is checking the timeSrv, which uses the requested duration to calculate when to show the zoom in button.

So I made the change to pass in the dashboard time into this panel instead of calculating the range based on response data:
<img width="1722" alt="image" src="https://github.com/grafana/grafana/assets/109082771/42790f4b-0001-41af-b1d7-a3ec19890faa">
I don't think we want this? 

There's still problems with this datasource on the backend, you're always going to be able to get it into the data outside range state even with the proposed change. For example, request more datapoints then the step of the query:
<img width="1725" alt="image" src="https://github.com/grafana/grafana/assets/109082771/8bb6a08e-6051-4f6e-a794-3391b9eac390">

as the backend doesn't seem to care about the request from, it only grabs the "to", and subtracts the query step until the max lines is hit.

TL;DR The testdata backend doesn't return data bounded by the requested range so there's always going to be certain situations where that scenario triggers the data outside range state.

**Comment by leeoniya** [comment]
probably relevant: we're discussing adding the queryTimeRange to the response metas for other purposes. can also be useful here.

https://github.com/grafana/grafana/pull/72715
